symphony-bdk-core/src/main/java/com/symphony/bdk/core/auth/impl/AbstractBotAuthenticator.java

```diff
@@ -15,22 +16,34 @@
 @API(status = API.Status.INTERNAL)
 public abstract class AbstractBotAuthenticator implements BotAuthenticator {
 
-  private final AuthenticationRetry<String> authenticationRetry;
+  private final AuthenticationRetry<String> kmAuthenticationRetry;
+  private final AuthenticationRetry<Token> podAuthenticationRetry;
 
   public AbstractBotAuthenticator(BdkRetryConfig retryConfig) {
-    authenticationRetry = new AuthenticationRetry<>(retryConfig);
+    kmAuthenticationRetry = new AuthenticationRetry<>(retryConfig);
+    podAuthenticationRetry = new AuthenticationRetry<>(retryConfig);
   }
 
   protected String retrieveToken(ApiClient client) throws AuthUnauthorizedException {
     final String unauthorizedMessage = String.format("Service account \"%s\" is not authorized to authenticate. "
         + "Check if credentials are valid.", getBotUsername());
 
-    return authenticationRetry.executeAndRetry("AbstractBotAuthenticator.retrieveToken", client.getBasePath(),
+    return kmAuthenticationRetry.executeAndRetry("AbstractBotAuthenticator.retrieveToken", client.getBasePath(),
         () -> authenticateAndGetToken(client), unauthorizedMessage);
   }
 
+  protected Token retrieveAuthToken(ApiClient client) throws AuthUnauthorizedException {
```

Comment:
if this fails do we automatically retry the token auth?

Reply:
yes retry is handled here https://github.com/finos/symphony-bdk-java/pull/618/files#diff-a931a6dbf82711870dce308eb96c3208300b8c0ec3373962225b7bc3ee522cd0R39